src/lib/TopLevel.hs

```diff
@@ -775,7 +778,13 @@ evalLLVM block = do
   resultVals <-
     liftIO $ callNativeFun nativeFun benchRequired logger [] resultTypes
   resultValsNoPtrs <- mapM litValToPointerlessAtom resultVals
-  applyNaryAbs reconAtom $ (map SubstVal resultValsNoPtrs :: [CAtomSubstVal AtomNameC n])
+  -- TODO: this unsafe coerce won't be needed once we treat everything as a
+  -- function, because then we won't need the reconstruction at all. If we were
```

Comment:
what does "once we treat everything as a function" mean?

Reply:
Specifically I mean unify the compilation paths around first-order monomorphic ("FOMO") functions, including top blocks, which get treated as nullary functions.